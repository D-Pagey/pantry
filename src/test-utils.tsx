import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';
import React, { FC, ReactNode } from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { FirebaseContext } from './components/ProviderFirebase';

type AllTheProvidersProps = {
    children: ReactNode;
    customContext?: any;
};

const AllTheProviders: FC<AllTheProvidersProps> = ({ children, customContext }) => (
    <FirebaseContext.Provider
        value={{
            categories: [],
            deleteFoodItem: () => () => null,
            expiringCount: 0,
            isAuthed: false,
            isCheckingAuth: false,
            fridge: [],
            signOut: () => null,
            updateFridge: () => null,
            user: {
                email: null,
                name: null
            },
            ...customContext
        }}
    >
        <MemoryRouter>{children}</MemoryRouter>
    </FirebaseContext.Provider>
);

const customRender = (ui: any, customContext?: any, options?: any) =>
    render(ui, { wrapper: () => <AllTheProviders customContext={customContext}>{ui}</AllTheProviders>, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
