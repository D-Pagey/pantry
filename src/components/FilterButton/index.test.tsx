import React from 'react';
import { render, screen } from '../../test-utils';
import { FilterButton } from '.';
import userEvent from '@testing-library/user-event';

const props = {
    onClick: () => null
};

describe('FilterButton component', () => {
    it('should render', () => {
        const { container } = render(<FilterButton {...props}>Vegetables</FilterButton>);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('should call onClick when X clicked', () => {
        const onClick = jest.fn();
        render(
            <FilterButton {...props} onClick={onClick}>
                Vegetables
            </FilterButton>
        );

        userEvent.click(screen.getByText('X'));

        expect(onClick).toHaveBeenCalled();
    });
});
