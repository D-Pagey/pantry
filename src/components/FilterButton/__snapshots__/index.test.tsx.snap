// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterButton component should render 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin: 0 0.5rem 0 0;
}

.c1 {
  background-color: #348841;
  color: white;
  padding: 0.5rem;
  border-radius: 10px 0 0 10px;
}

.c2 {
  border: none;
  background: #EB5757;
  border-radius: 0 10px 10px 0;
  cursor: pointer;
  color: white;
  padding: 0 0.4rem 0 0.3rem;
}

<div
  class="c0"
>
  <span
    class="c1"
  >
    Vegetables
  </span>
  <button
    class="c2"
    type="button"
  >
    X
  </button>
</div>
`;
